### PR TITLE
Mixpanel: track diagram deployments

### PR DESCRIPTION
**Pull request by smbea**
Related to #2934

### CODE REVIEW

File: client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+      const {
+        contents
+      } = file;
+
+      if (!diagramType) {
+        return;
+      }
+
+      const baseEvent = context === 'deploymentTool' ? 'deploy' : 'startInstance';
```
Comment (pinussilvestrus) [inline client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js:80]: Just for my understanding: This differs from [the way we track it in ET](https://docs.camunda.io/docs/components/modeler/desktop-modeler/telemetry/#deployment-event) (where we have all this inside the `deployment` context). I guess this is intended due to how Mixpanel works with the event names?

File: client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+/**
```
Comment (pinussilvestrus) [inline client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js:1]: Do we already track to update our documentation regarding this tracking? I can't find something related in the e[pic issue](https://github.com/camunda/camunda-modeler/issues/2934).

**Review by pinussilvestrus — COMMENTED** [review]
Implementation looks good to me, just some understanding questions 👍

File: client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js (hunk 1 above)
Comment (smbea) [reply client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js:80]: Yes, we have a different naming strategy for Mixpanel

File: client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js (hunk 2 above)
Comment (smbea) [reply client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js:1]: It's actually being tracked here: https://github.com/bpmn-io/internal-docs/issues/504

Comment (pinussilvestrus) [reply client/src/plugins/user-journey-statistics/event-handlers/DeploymentEventHandler.js:1]: Oh that's the epic, nice 🙂 Thanks!

**Review by pinussilvestrus — APPROVED** [review]
LGTM

**Pull request merged**